### PR TITLE
Update remoteRendered to listen for 404s and return the correct status code

### PR DESCRIPTION
**Pull request by gtrufitt**
## What does this change?

In https://github.com/guardian/dotcom-rendering/pull/1618 we started returning 404s from DCR to Frontend, in order to force an uncached 'Not Found' error into the AMP cache and serve directly from Dotcom. 

## Does this change need to be reproduced in dotcom-rendering ?

- [x] Yes https://github.com/guardian/dotcom-rendering/pull/1618 

## Screenshots

image

### Does this affect other platforms?

- [X] AMP 

### Tested

- [X] Locally
- [X] On CODE (optional)

### CODE REVIEW

File: article/app/renderers/RemoteRenderer.scala [hunk 1]
```diff
@@ -46,6 +50,9 @@ class RemoteRenderer extends Logging {
             case 200 =>
               Cached(article)(RevalidatableResult.Ok(Html(response.body)))
                 .withHeaders("X-GU-Dotcomponents" -> "true")
+            case 404 =>
+              Cached(CacheTime.NotFound)(WithoutRevalidationResult(NotFound))
```
Comment (jfsoul) [inline article/app/renderers/RemoteRenderer.scala:54]: I saw somewhere else earlier where 404 wasn't cached (after CAPI returns 404); thinking about it I suppose it doesn't make it back directly to fastly from here to be cached anyway. Not sure whether it'd be clearer to cache or not in this case.

**Review by jfsoul — APPROVED** [review]

**Comment by shtukas** [comment]
Comes together with: https://github.com/guardian/dotcom-rendering/pull/1618

**Review by shtukas — APPROVED** [review]

**Comment by mxdvl** [comment]
Hi there!

Should this PR remain open? It's been around for a while and should probably either ship or be closed.
If it needs to remain open, could you please change the base branch to main? See mention above for context.

🧹💨 @gtrufitt @shtukas @jfsoul

**Comment by gtrufitt** [comment]
> Hi there!
> 
> Should this PR remain open? It's been around for a while and should probably either ship or be closed.
> If it needs to remain open, could you please change the base branch to main? See mention above for context.
> 
> 🧹💨 @gtrufitt @shtukas @jfsoul

Updated 👍

**Pull request closed without merging**